fuuk/people/models/grant.py

```diff
@@ -22,31 +25,62 @@ def __unicode__(self):
+    editor = models.ForeignKey(User, blank=True, null=True, on_delete=models.SET_NULL, verbose_name=_('editor'))
+    investigator_prefix = models.CharField(_('prefix'), max_length=20, blank=True, null=True)
+    investigator_first_name = models.CharField(_('first name'), max_length=50)
+    investigator_last_name = models.CharField(_('last name'), max_length=50)
+    investigator_suffix = models.CharField(_('suffix'), max_length=20, blank=True, null=True)
+    investigator_institution = models.ForeignKey(Institution, blank=True, null=True, on_delete=models.CASCADE,
+                                                 verbose_name=_('institution'))
+    investigator_human = models.ForeignKey(Human, blank=True, null=True, on_delete=models.SET_NULL,
+                                           verbose_name=_('human'))
+    number = models.CharField(_('grant number'), max_length=20)
+    start = models.SmallIntegerField(_('start year'), validators=[MinValueValidator(1990),
+                                     MaxValueValidator(date.today().year + 1)])
+    end = models.SmallIntegerField(_('end year'), validators=[MinValueValidator(1990),
+                                   MaxValueValidator(date.today().year + 10)])
+    agency = models.ForeignKey(Agency, help_text=_('Contact administrators for different Grant Agency.'),
+                               verbose_name=_('grant agency'))
 
-    title = models.CharField(max_length=200)
-    annotation = models.TextField(help_text=get_markdown_help_text)
+    title = models.CharField(_('title'), max_length=200)
+    annotation = models.TextField(_('grant annotation'), help_text=get_markdown_help_text)
 
     class Meta:
         app_label = 'people'
         unique_together = (('number', 'agency'),)
+        verbose_name = _('grant')
+        verbose_name_plural = _('grants')
 
     def __unicode__(self):
         return self.title or u""
 
-    def clean(self):
-        # Many-to-many fields can only be checked if instance is already saved
-        if self.pk and self.co_authors.all():
-            # Check human if possible
-            if self.author.human:
-                if self.author.human in self.co_authors.values_list('human', flat=True):
-                    raise ValidationError('Author can not be also co_author.')
-            else:
-                if self.author in self.co_authors.all():
-                    raise ValidationError('Author can not be also co_author.')
+    @property
+    def investigator_name(self):
+        return u"%s %s" % (self.investigator_last_name, self.investigator_first_name)
+    investigator_name.fget.short_description = _('applicant name')
+
+    @property
+    def investigator_full_name(self):
+        return full_name(self.investigator_prefix, self.investigator_first_name, self.investigator_last_name,
+                         self.investigator_suffix)
+    investigator_full_name.fget.short_description = _('applicant full name')
```

Comment:
The `fget` is necessary if I want the function with `@property` decorator and not using `property()` function, which introduces new variable to the model:
https://github.com/django/django/commit/cec9558fba1bc6401ea2ec6d71b816b4dfd31b28